.github/pull_request_template.md

```diff
@@ -1,18 +1,86 @@
-Describe your changes here.
+## Summary
 
-Closes # (type the issue number after # if applicable; otherwise remove this line)
+<!--
+A few sentences describing the changes being proposed in this pull request.
+-->
 
-### Screenshots
+## List of notable changes
 
-Please provide before/after screenshots for any visual changes
+<!--
+E.g.
 
-### Merge checklist
+- **added** # for # component because #
+- **removed** props for # component because #
+- **updated** documentation for # component because #
+-->
 
-- [ ] Added/updated tests
-- [ ] Added/updated documentation
-- [ ] Tested in Chrome
-- [ ] Tested in Firefox
-- [ ] Tested in Safari
-- [ ] Tested in Edge
+-
+-
 
-Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.
+## What should reviewers focus on?
+
+-
+-
+
+## Steps to test
+
+<!--
+Help reviewers test the feature by providing steps to reproduce the behavior.
+
+E.g.
+
+1. Open the # component in CI-deployed preview environment
+2. Go to # story in Storybook
+3. Verify that # behaves as described in the following issue.
+-->
+
+1.
+1.
+1.
+
+## Supporting resources (related issues, external links, etc)
+
+-
+-
+
+## Contributor checklist
+
+- [ ] All new and existing CI checks pass
+- [ ] Tests prove that the feature works and covers both happy and unhappy paths
+- [ ] Documentation has been added or updated if necessary
+- [ ] Any drop in coverage, breaking changes, or regressions have been documented in the PR description
+- [ ] New visual snapshots have been generated or updated for any UI changes
+- [ ] New and existing behaviors work accross browsers (e.g., Chrome, Safari, Firefox, Edge)
+- [ ] All developer debugging and non-functional logging has been removed
+- [ ] Related issues have been referenced in the PR description
+- [ ] A [changeset](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) has been added to describe any consumer-facing changes in the PR. Include migration steps for any breaking changes.
+
+
+## Reviewer checklist
+
+- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../contributor-docs/CONTRIBUTING.md) and [code of conduct](../../contributor-docs/CODE_OF_CONDUCT.md)
+- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
+- [ ] Check that documentation is up-to-date
+- [ ] Check that there aren't other open Pull Requests for the same update/change
```

Comment:
I feel like these sections would be more helpful under a "definition of done" ADR/document or as items within the issue itself that is being closed.